js/src/Scatter.js

```diff
@@ -729,6 +791,31 @@ define(["d3", "./Mark", "./utils", "./Markers", "underscore"],
             this.apply_styles();
         },
 
+        update_hovered: function(model, value) {
+            this.hovered_index = value;
+            this.apply_styles();
+        },
+
+        apply_styles: function() {
+            var all_indices = _.range(this.model.mark_data.length);
+            this.clear_style(this.selected_style);
+            this.clear_style(this.unselected_style);
+            this.clear_style(this.hovered_style);
+	    this.clear_style(this.unhovered_style);
+
+            this.set_default_style(all_indices);
+
+            this.set_style_on_elements(this.selected_style, this.selected_indices);
+            var unselected_indices = (!this.selected_indices) ?
+                [] : _.difference(all_indices, this.selected_indices);
+            this.set_style_on_elements(this.unselected_style, unselected_indices);
```

Comment:
Probably everything before this can go into a super call. What do you think @ssunkara1 ?